x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engines_lens/lens_data.ts

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { DataView } from '@kbn/data-views-plugin/public';
+import { i18n } from '@kbn/i18n';
+import {
+  DateHistogramIndexPatternColumn,
+  FormulaPublicApi,
+  PersistedIndexPatternLayer,
+  TypedLensByValueInput,
+} from '@kbn/lens-plugin/public';
+import { MetricVisualizationState } from '@kbn/lens-plugin/public/visualizations/metric/visualization';
+
+export const queries = {
+  chartColor: '#6092C0',
+  label: i18n.translate(
+    'xpack.enterpriseSearch.content.engine.overview.analytics.lens.queries.chart.title',
+    {
+      defaultMessage: 'Queries',
+    }
+  ),
+};
+export const withNoResults = {
+  chartColor: '#54B399',
+  label: i18n.translate(
+    'xpack.enterpriseSearch.content.engine.overview.analytics.lens.withNoResults.chart.title',
+    {
+      defaultMessage: 'No Results',
+    }
+  ),
+};
+enum formula {
+  totalQueries = 'count(event.customer_data.query)',
+  totalQueriesWithNoResults = "count(kql='event.customer_data.totalResults :0')",
+  totalQueriesPercentageChange = "(count(event.customer_data.query) - count(event.customer_data.query, shift='previous')) / abs(count(event.customer_data.query, shift='previous'))",
+  totalQueriesPercentageChangeWithNoResults = "(count(kql='event.customer_data.totalResults :0') - count(kql='event.customer_data.totalResults :0', shift='previous')) / abs(count(kql='event.customer_data.totalResults :0', shift='previous'))",
+}
+const metricFormula = (isShowingTotalQueries: boolean) => {
+  return isShowingTotalQueries
+    ? [formula.totalQueries, formula.totalQueriesPercentageChange]
+    : [formula.totalQueriesWithNoResults, formula.totalQueriesPercentageChangeWithNoResults];
+};
+const xyFormula = (isShowingTotalQueries: boolean) => {
+  return isShowingTotalQueries ? formula.totalQueries : formula.totalQueriesWithNoResults;
+};
+
+export const getIcon = (percentage: number): string => {
+  return percentage >= 0 ? 'sortUp' : 'sortDown';
+};
+
+export const getLensXYLensAttributes = (
+  dataView: DataView | null,
+  formulaApi: FormulaPublicApi,
+  isShowingTotalQueries: boolean
+): TypedLensByValueInput['attributes'] => {
+  const dataLayer = formulaApi.insertOrReplaceFormulaColumn(
+    'col2',
+    {
+      formula: xyFormula(isShowingTotalQueries),
+      label: isShowingTotalQueries ? queries.label : withNoResults.label,
+    },
+    {
+      columnOrder: ['col1'],
+      columns: {
+        col1: {
```

Comment:
suggestion: naming these columns to be more descriptive and having an enum for this.